app/ui/legacy/src/main/java/com/fsck/k9/ui/account/AccountsViewModel.kt

```diff
@@ -1,8 +1,118 @@
+        callbackFlow {
+            Timber.d("getAccountsFlow.callbackFlow")
+
+            send(preferences.accounts)
+
+            Timber.d("sent.accounts")
+
+            val accountsChangeListener = AccountsChangeListener {
+                Timber.d("onAccountsChanged")
+                sendBlocking(preferences.accounts)
+            }
+            preferences.addOnAccountsChangeListener(accountsChangeListener)
+            awaitClose {
+                Timber.d("removeOnAccountsChangeListener")
+                preferences.removeOnAccountsChangeListener(accountsChangeListener)
+            }
+        }
+
+    private val displayAccountFlow: Flow<List<DisplayAccount>> = accountsFlow
+        .flatMapLatest { accounts ->
+            Timber.d("displayAccountFlow.flatMapLatest; accounts: %s", accounts)
+
+            val unreadCountFlows: List<Flow<Int>> = accounts.map { account ->
+                Timber.d("account: %s", account.uuid)
+                getUnreadCountFlow(account)
+            }
+
+            combine(unreadCountFlows) { unreadCounts ->
+                Timber.d("unreadCounts: %s", unreadCounts)
+                unreadCounts.mapIndexed { index, unreadCount ->
+                    Timber.d("index, unreadCount: %d, %s", index, unreadCounts)
+                    DisplayAccount(account = accounts[index], unreadCount)
+                }
+            }
+        }
+
+    private fun getUnreadCountFlow(account: Account): Flow<Int> {
+        return callbackFlow {
+            Timber.d("getUnreadCountFlow.callbackFlow; account: %s", account)
+
+            val notificationUri = EmailProvider.getNotificationUri(account.uuid)
+
+            viewModelScope.launch(Dispatchers.IO) {
+                send(unreadMessageCountProvider.getUnreadMessageCount(account))
+            }
+
+            val contentObserver = object : ContentObserver(Handler()) {
+                override fun onChange(selfChange: Boolean) {
+                    Timber.d("selfChange: %s", selfChange)
+                    viewModelScope.launch(Dispatchers.IO) {
+                        sendBlocking(unreadMessageCountProvider.getUnreadMessageCount(account))
+                    }
+                }
+            }
+
+            Timber.d("registerContentObserver")
+            contentResolver.registerContentObserver(notificationUri, false, contentObserver)
+
+            awaitClose {
+                Timber.d("unregisterContentObserver")
+                contentResolver.unregisterContentObserver(contentObserver)
+            }
+        }
+    }
+
+    init {
+        Timber.d("init")
+        viewModelScope.launch {
+            Timber.d("viewModelScope.launch")
+            displayAccountFlow
+                .collect {
+                    Timber.d("mutableAccountAndUnreadsLiveData.value")
+                    mutableAccountAndUnreadsLiveData.value = it
+                }
+            Timber.d("finished collect call")
+        }
+    }
```

Comment:
To convert a `Flow` to `LiveData` we can use the `asLiveData()` extension function.

```kotlin
val displayAccountsLiveData: LiveData<List<DisplayAccount>> = displayAccountFlow.asLiveData()
```